packages/react/src/primitives/Pagination/usePaginationItems.tsx

```diff
@@ -39,7 +40,7 @@ export const usePaginationItems = (
       key="next"
       currentPage={currentPage}
       onClick={onNext}
-      isDisabled={currentPage >= totalPages}
+      isDisabled={currentPage === totalPages && !hasMorePages}
```

Comment:
This is the only core logic added to the Pagination component in this PR

Reply:
This is `currentPage === totalPages` should be good enough theoretically. But remember `currentPage` was passed in by customers so it may not always make sense. I forget if I handle `currentPage` > `totalPages` in the hooks. But this is the context why I use `>=` here.